### PR TITLE
First Lesson

### PR DESCRIPTION
**Pull request by dakruchko**

### CODE REVIEW

File: sites/all/modules/custom/lesson_menu/lesson_menu.info [hunk 1]
```diff
@@ -0,0 +1,3 @@
+name = Lesson Menu Module
+description = Hometask #1 module
+core = 7.x
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.info]: Add group

Comment (dwebpoint) [reply sites/all/modules/custom/lesson_menu/lesson_menu.info]: Add empty line

File: sites/all/modules/custom/lesson_menu/lesson_menu.module [hunk 2]
```diff
@@ -0,0 +1,285 @@
+<?php
+
+/**
+ * @file
+ * Describes functions to create custom menu items, handle arguments and customize theme
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.module]: Add dot at the end of the line

File: sites/all/modules/custom/lesson_menu/lesson_menu.module [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+ * @file
+ * Describes functions to create custom menu items, handle arguments and customize theme
+ */
+
+/**
+ * Implements hook_menu().
+ */
+function lesson_menu_menu() {
+  $items['training/menu'] = [
+    'title' => 'Training Landing Page',
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.module:13]: Add description

File: sites/all/modules/custom/lesson_menu/lesson_menu.module [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+    'page callback' => 'lesson_menu_render',
+    'access arguments' => [
+      'access content',
+    ],
+    'type' => MENU_NORMAL_ITEM,
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.module]: Delete the default value

File: sites/all/modules/custom/lesson_menu/lesson_menu.module [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+  return $items;
+}
+
+/**
+ * Implements hook_permission().
+ */
+function lesson_menu_permission() {
+  $new_permission = [
+    'access protected menu example' => [
+      'title' => t('access protected menu example'),
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.module]: Add description

File: sites/all/modules/custom/lesson_menu/lesson_menu.module [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+  ];
+  return $items;
+}
+
+/**
+ * Implements hook_permission().
+ */
+function lesson_menu_permission() {
+  $new_permission = [
+    'access protected menu example' => [
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.module]: Use module name

File: sites/all/modules/custom/lesson_menu/lesson_menu.module [hunk 7]
```diff
@@ -0,0 +1,285 @@
...
+    'access arguments' => [
+      'access content',
+    ],
+    'type' => MENU_NORMAL_ITEM,
+    'expanded' => TRUE,
+    'file' => 'lesson_menu.pages.inc',
+  ];
+  $items['training/menu/custom_access'] = [
+    'title' => 'Custom Access Menu Item',
+    'access callback' => 'lesson_menu_user_authenticated',
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.module]: Use user_is_logged_in()

File: sites/all/modules/custom/lesson_menu/lesson_menu.module [hunk 8]
```diff
@@ -0,0 +1,285 @@
...
+ */
+function lesson_menu_user_authenticated() {
+  $account = $GLOBALS['user'];
+  if (in_array('authenticated user', $account->roles)) {
+    return TRUE;
+  }
+}
+
+/**
+ * _load function to handle and prepare passed argument
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.module]: Comment

File: sites/all/modules/custom/lesson_menu/lesson_menu.module [hunk 9]
```diff
@@ -0,0 +1,285 @@
...
+ */
+function two_load($val) {
+  $two = $val;
+  return $two;
+}
+
+/**
+ * _load function to handle and prepare passed argument
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.module]: Comment

File: sites/all/modules/custom/lesson_menu/lesson_menu.module [hunk 10]
```diff
@@ -0,0 +1,285 @@
...
+ */
+function argument_load($val) {
+  switch ($val) {
+    case 1:
+      $val = 'one';
+      break;
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.module]: Empty line after break;

File: sites/all/modules/custom/lesson_menu/lesson_menu.module [hunk 11]
```diff
@@ -0,0 +1,285 @@
...
+  return $two;
+}
+
+/**
+ * _load function to handle and prepare passed argument
+ */
+function argument_load($val) {
+  switch ($val) {
+    case 1:
+      $val = 'one';
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.module]: Add t()

File: sites/all/modules/custom/lesson_menu/lesson_menu.module [hunk 12]
```diff
@@ -0,0 +1,285 @@
...
+    case 3:
+      $val = 'three';
+      break;
+    case 99:
+      $val = 'Almost there';
+      break;
+    default:
+      $val = '404';
+  }
+  return $val;
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.module]: [Optional] Add empty line before return

File: sites/all/modules/custom/lesson_menu/lesson_menu.module [hunk 13]
```diff
@@ -0,0 +1,285 @@
...
+}
+
+/**
+ * Implements hook_theme().
+ */
+function lesson_menu_theme() {
+  $theme['render_function'] = [
+    'template' => 'templates/my_template'
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.module]: Change template name, theme function name

File: sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc [hunk 14]
```diff
@@ -0,0 +1,191 @@
...
+function lesson_menu_render() {
+  return t('This is the landing page of the Training module');
+}
+
+/**
+ * Page callback: Displays a tab page.
+ *
+ * @see lesson_menu_menu()
+ */
+function lesson_menu_render_tabs($index) {
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc]: Refactor function

File: sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc [hunk 15]
```diff
@@ -0,0 +1,191 @@
...
+function lesson_menu_render_div($argument) {
+  return '<div>' . $argument . '</div>';
+}
+
+/**
+ * Page callback: Displays a page with prepared argument.
+ *
+ * @see lesson_menu_menu()
+ */
+function lesson_menu_render_handle_argument($argument) {
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc]: Function name

File: sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc [hunk 16]
```diff
@@ -0,0 +1,191 @@
...
+  }
+}
+
+/**
+ * Page callback: Displays a page with visitor's name.
+ *
+ * @see lesson_menu_menu()
+ */
+function lesson_menu_username() {
+  $account = $GLOBALS['user'];
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc]: global $user;

File: sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc [hunk 17]
```diff
@@ -0,0 +1,191 @@
...
+  return 'Dynamic title: username=' . $output;
+}
+
+/**
+ * Page callback: Displays a page with list of links.
+ *
+ * @see lesson_menu_menu()
+ */
+function lesson_menu_render_theme() {
+  $pages = [
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc]: theme_link

File: sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc [hunk 18]
```diff
@@ -0,0 +1,191 @@
...
+}
+
+/**
+ * Page callback: Prepares data and passes it to rendering.
+ *
+ * @see lesson_menu_menu()
+ */
+function lesson_menu_render_image() {
+  $attributes = [
+    'class' => ['simple-image', ],
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc]: remove ,

File: sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc [hunk 19]
```diff
@@ -0,0 +1,191 @@
...
+  ];
+
+  if (!file_exists('public://images/sample.png')) {
+    copy('modules/image/sample.png', 'public://images/sample.png');
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc]: Find Drupal way how the file can be copied

File: sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc [hunk 20]
```diff
@@ -0,0 +1,191 @@
...
+    $styles[] = $k;
+  }
+
+  $list_of_styles = theme('item_list', $variables = [
+    'items' => $styles,
+    'type' => 'ol',
+    'title' => 'Images Styles',
+  ]);
+
+  return theme('render_function', $variables = [
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc]: Add variables to theme function

File: sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc [hunk 21]
```diff
@@ -0,0 +1,191 @@
...
+  ]);
+}
+
+/**
+ * Page callback: Displays a page with content that depends on user's permission.
+ *
+ * @see lesson_menu_menu()
+ */
+function lesson_menu_render_access() {
+  if (user_access('access protected content') === TRUE) {
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/lesson_menu.pages.inc]: Remove ===True

File: sites/all/modules/custom/lesson_menu/templates/my_template.tpl.php [hunk 22]
```diff
@@ -0,0 +1,22 @@
...
+ *
+ * @see lesson_menu_render_image()
+ *
+ */
+
+print $original_image;
+
+print $list;
+
+foreach ($generated_images as $k => $v) {
```
Comment (dwebpoint) [inline sites/all/modules/custom/lesson_menu/templates/my_template.tpl.php]: Refactor
https://api.drupal.org/api/drupal/includes%21theme.inc/function/theme_image/7.x

**Comment by dakruchko** [comment]
1

**Pull request closed without merging**